### PR TITLE
Survey

### PR DESCRIPTION
**Pull request by Sushilji**
Working code of Survey branch with user role, survey db and insert script for admins in "dBScript" folder

### CODE REVIEW

**Review by smartdusttechnologies — COMMENTED** [review]
1. abcEmail.cshtml, abcAdmin.cshtml, why it is prefixed with abc?
2. "UserTemplet": "emailformat/abcmail.html",
    "AdminMail": "emailformat/abcAdminMail.html",
3. put it in separate file:
public enum Testing
    {
        Testing, 
        Calibration, 
        Other
    }
5. Email service is written just to work with Survey?
mm.Subject = "Survey"; and passed the model survey only

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/wwwroot/emailformat/TestingAndCalibrationsSurveyMailTemplate.html [hunk 1]
```diff
@@ -0,0 +1,149 @@
+<!DOCTYPE html>
+<html lang="en" xmlns:o="urn:schemas-microsoft-com:office:office" xmlns:v="urn:schemas-microsoft-com:vml">
+
+<head>
+    <title></title>
+    <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
+    <meta content="width=device-width,initial-scale=1" name="viewport" />
+    <!--[if mso]><xml><o:OfficeDocumentSettings><o:PixelsPerInch>96</o:PixelsPerInch><o:AllowPNG/></o:OfficeDocumentSettings></xml><![endif]-->
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/wwwroot/emailformat/TestingAndCalibrationsSurveyMailTemplate.html:8]: is this commented lines are useful?

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Testing.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace TestingAndCalibrationLabs.Web.UI.Models
+{
+    /// <summary>
+    /// Types of Test
+    /// </summary>
+    public enum Testing
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Testing.cs:11]: plz rename the file name and enum to TestingType.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/SurveyModel.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        /// Email address
+        /// </summary>
+        [Required(ErrorMessage = "Please enter your email address")]
+        [DataType(DataType.EmailAddress)]
+        [Display(Name = "Email address")]
+        [MaxLength(50)]
+        [RegularExpression("[a-z0-9._%+-]+@[a-z0-9.-]+.[a-z]{2,4}", ErrorMessage = "Please enter correct email")]
+
+        //[Required, EmailAddress(ErrorMessage ="Please enter correct email id"]
+        public string Email { get; set; }
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/SurveyModel.cs:36]: remove commented code.

**Review by smartdusttechnologies — COMMENTED** [review]
is these files are needed to be check-in:
TestingAndCalibrationLabs/TestingAndCalibrationLabs.DB/TestingAndCalibrationLabs.DB.dbmdl
TestingAndCalibrationLabs/TestingAndCalibrationLabs.DB/TestingAndCalibrationLabs.DB.jfm

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Interfaces/ISurveyService.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+using System.Text;
+using TestingAndCalibrationLabs.Business.Common;
+using TestingAndCalibrationLabs.Business.Core.Model;
+
+namespace TestingAndCalibrationLabs.Business.Core.Interfaces
+{
+    public interface ISurveyService
+    {
+        RequestResult<int> Add(SurveyModel mailsend);
+        bool Sendemail(EmailModel emailModel);
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Interfaces/ISurveyService.cs:12]: why survey also has sendEmail method?

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Testing.cs (hunk 2 above)
Comment (Sushilji) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Testing.cs:11]: Changed Testing---> TestingType

But "enum" cannot be changed to "TestingType" it produces error

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/TestingAndCalibrationService/SurveyService.cs [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+            _userRepository = userRepository;
+            _configuration = configuration;
+            _hostingEnvironment = hostingEnvironment;
+        }
+        public RequestResult<int> Add(SurveyModel surveymodel)
+        {
+            var emailsendTo = _userRepository.Get();
+
+            //Read other values from Appsetting .Json 
+            surveymodel.EmailTemplate = _configuration["TestingAndCalibrationSurvey:UserTeplet"];
```
[the remark this answers is not in the document]
Comment (Sushilji) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/TestingAndCalibrationService/SurveyService.cs:40]: done

**Comment by Sushilji** [comment]
Done

**Review by smartdusttechnologies — APPROVED** [review]
looks good

**Pull request merged**